services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/OVSGoalStateServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.service.impl;
+
+import com.futurewei.alcor.common.message.MessageClient;
+import com.futurewei.alcor.dataplane.config.Config;
+import com.futurewei.alcor.dataplane.config.grpc.GoalStateProvisionerClient;
+import com.futurewei.alcor.dataplane.service.GoalStateService;
+import com.futurewei.alcor.schema.Goalstate;
+import com.futurewei.alcor.schema.Goalstateprovisioner;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class OVSGoalStateServiceImpl implements GoalStateService {
+  private int port;
+  private Goalstate.GoalState goalState;
```

Comment:
A service should be able to handle multiple GS messages. 

Why is a GS implementation tied to a GS message here?

Reply:
service IS able to handle multiple in parallel, we would use multiple service instance in thread level to make sure performance and secure, if only use service handle multiple GS messages which is later on done on the blockingstub way of grpc, it would anyway block others due to thread safety

Reply:
changed in 5dee85b per discussion